websites/Y/YouTube/presence.ts

```diff
@@ -424,11 +438,24 @@ presence.on("UpdateData", async () => {
         presenceData.startTimestamp = browsingStamp;
       }
       if (channelPic) {
-        presenceData.largeImageKey = document
-          .querySelector<HTMLImageElement>(
-            "#avatar.ytd-c4-tabbed-header-renderer > img"
-          )
-          .src.replace("=s176", "=s512");
+        const channelImg =
+          document //self channel
+            .querySelector<HTMLImageElement>(
+              "yt-img-shadow.ytd-channel-avatar-editor > img"
+            )
+            ?.src.replace("=s176", "=s512") ??
+          document
+            .querySelector<HTMLImageElement>(
+              "#avatar.ytd-c4-tabbed-header-renderer > img"
+            )
+            ?.src.replace("=s176", "=s512") ??
+          document //when viewing a community post
+            .querySelector<HTMLImageElement>(
+              "#author-thumbnail > a > yt-img-shadow > img"
+            )
+            ?.src.replace("=s76", "=s512") ??
+          "yt_lg";
+        if (channelImg !== "") presenceData.largeImageKey = channelImg;
```

Comment:
An empty string is falsey but test it first pelase 
```suggestion
        if (!channelImg) presenceData.largeImageKey = channelImg;
```

Reply:
this doesn't work

Reply:
I didn't read it properly, try this: 
```suggestion
        if (channelImg) presenceData.largeImageKey = channelImg;
```

Reply:
Oh ye now it does